llama-index-integrations/llms/llama-index-llms-gaudi/llama_index/llms/gaudi/base.py

```diff
@@ -0,0 +1,385 @@
+# This file is adapted from
+# https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/
+# llms/llama-index-llms-huggingface/llama_index/llms/huggingface/base.py
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Any, Callable, List, Optional, Sequence, Union
+
+import torch
+from llama_index.core.base.llms.types import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+)
+from llama_index.core.bridge.pydantic import Field, PrivateAttr
+from llama_index.core.callbacks import CallbackManager
+from llama_index.core.constants import (
+    DEFAULT_CONTEXT_WINDOW,
+    DEFAULT_NUM_OUTPUTS,
+)
+from llama_index.core.llms.callbacks import (
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.core.llms.custom import CustomLLM
+
+from llama_index.core.base.llms.generic_utils import (
+    completion_response_to_chat_response,
+    stream_completion_response_to_chat_response,
+    messages_to_prompt as generic_messages_to_prompt,
+)
+from llama_index.core.prompts.base import PromptTemplate
+from llama_index.core.types import BaseOutputParser, PydanticProgramMode, Thread
+from transformers import (
+    StoppingCriteria,
+    StoppingCriteriaList,
+)
+from transformers import AutoTokenizer, LlamaTokenizer
+#gaudi
+from llama_index.llms.gaudi.utils import initialize_model
+from llama_index.llms.huggingface import HuggingFaceLLM
+
+#DEFAULT_HUGGINGFACE_MODEL = "meta-llama/Llama-2-7b-chat-hf"
+DEFAULT_HUGGINGFACE_MODEL = "/home/ubuntu/jean/models/mistral"
+
+logger = logging.getLogger(__name__)
+
+
+class GaudiLLM(CustomLLM):
```

Comment:
This seems like mostly a copy of HuggingFaceLLM ? Is there any reason that a user can't directly use a HuggingFaceLLM with these models? If its just a difference in the `init`, should we just subclass `HuggingFaceLLM` to reduce code duplication?

Reply:
I agree with you. Updated base.py to use HuggingFaceLLM as base class.